packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -966,11 +966,6 @@ export function cloneAndReplaceKey(oldElement, newKey) {
     __DEV__ && enableOwnerStacks ? oldElement._debugStack : undefined,
     __DEV__ && enableOwnerStacks ? oldElement._debugTask : undefined,
   );
-  if (__DEV__) {
-    // The cloned element should inherit the original element's key validation.
-    clonedElement._store.validated = oldElement._store.validated;
```

Comment:
If the original element was `validated === 2` because it came from Flight or another Children helper, then was passed through map again, then it turns into `validated === 0` but since it does have a key now it wouldn't issue an error even though it should've.

Reply:
When I started tackling this, I was also considering copying this logic to `React.cloneElement`.

I'll revert this change for now, but I mention it to ask whether you think `React.cloneElement` should also persist the original element's `_store.validated` (and whether that is affected by a new `key` being supplied to `React.cloneElement`).

If you think there's something to do there, I'd be happy to follow up.